d2/src/main/java/com/linkedin/d2/balancer/LastSeenBalancerWithFacilitiesFactory.java

```diff
@@ -73,15 +75,21 @@ public LoadBalancerWithFacilities create(D2ClientConfig config)
     }
 
     // init all the stores
-    LastSeenZKStore<ClusterProperties> lsClusterStore = getClusterPropertiesLastSeenZKStore(config, zkPersistentConnection, d2ClientJmxManager);
+    LastSeenZKStore<ClusterProperties> lsClusterStore =
+      getClusterPropertiesLastSeenZKStore(config, zkPersistentConnection, d2ClientJmxManager,
+                                          config._executorService, config.zookeeperReadWindowMs);
```

Comment:
just for my understanding, which config is this executor service coming from? where is it defined?

Reply:
This is coming from R2SchedulerFactory in container